Source/WebCore/rendering/RenderBox.cpp

```diff
@@ -1605,10 +1606,14 @@ void RenderBox::paintBoxDecorations(PaintInfo& paintInfo, const LayoutPoint& pai
     // The theme will tell us whether or not we should also paint the CSS background.
     bool borderOrBackgroundPaintingIsNeeded = true;
     if (style().hasEffectiveAppearance()) {
-        ControlStates* controlStates = controlStatesForRenderer(*this);
-        borderOrBackgroundPaintingIsNeeded = theme().paint(*this, *controlStates, paintInfo, paintRect);
-        if (controlStates->needsRepaint())
-            view().scheduleLazyRepaint(*this);
+        if (auto* control = theme().controlPartForRenderer(*this))
```

Comment:
I was going to ask why we can't put everything in the new `paint`, but then it would be tricker to distinguish whether we're going down the new path and also set `borderOrBackgroundPaintingIsNeeded`. I think this is fine.